src/librustc/dep_graph/graph.rs

```diff
@@ -1021,25 +997,17 @@ impl CurrentDepGraph {
     fn complete_anon_task(&mut self, kind: DepKind, task_deps: TaskDeps) -> DepNodeIndex {
         debug_assert!(!kind.is_eval_always());
 
-        let mut fingerprint = self.anon_id_seed;
         let mut hasher = StableHasher::new();
 
-        for &read in task_deps.reads.iter() {
-            let read_dep_node = self.data[read].node;
+        task_deps.reads.hash(&mut hasher);
```

Comment:
This PR changes the hashing of anon tasks to just hash the `DepNodeIndex`s for dependencies instead of the full `DepNode`s. This should be fine since the hash is supposed to be unique per session.